storage/local/persistence.go

```diff
@@ -606,7 +608,9 @@ func (p *persistence) checkpointSeriesMapAndHeads(fingerprintToSeries *seriesMap
 			if err != nil {
 				return
 			}
-			w.Write(buf)
+			if _, err = w.Write(buf); err != nil {
```

Comment:
@fabxc Oh, I didn't notice that. Doesn't that also mean that the whole return value never gets set properly if we're then continuously always using the overwritten `err`?